packages/firestore/src/core/target.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DocumentKey } from '../model/document_key';
+import { ResourcePath } from '../model/path';
+import { isNullOrUndefined } from '../util/types';
+import { Bound, Filter, OrderBy, Query } from './query';
+
+/**
+ * Target is the class we use to convert to query proto, to save in LocalStore
+ * and send to backend via RemoteStore. It only has the query attributes backend
+ * supports, unlike `Query`, which might have attributes we support locally
+ * in the SDK.
+ */
+export class Target {
+  private memoizedCanonicalId: string | null = null;
+
+  /**
+   * Initializes a Target with a path and optional additional query constraints.
+   * Path must currently be empty if this is a collection group query.
+   *
+   * NOTE: you should always construct `Target` from `Query.toTarget` instead of
+   * using this constructor, because `Query` provides sensible default `orderBy`
```

Comment:
What is "sensible"? In the rest of the code base we differentiate between "implicit" and "explicit". Can we use the same wording here?

Reply:
Done.